src/Common/src/Interop/OSX/Interop.CoreFoundation.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+using CFStringRef = System.IntPtr;
+using CFArrayRef = System.IntPtr;
+using FSEventStreamRef = System.IntPtr;
+using size_t = System.IntPtr;
+using FSEventStreamEventId = System.UInt64;
+using CFTimeInterval = System.Double;
+using CFRunLoopRef = System.IntPtr;
+
+internal static partial class Interop
+{
+    internal static partial class CoreFoundation
+    {
+        /// <summary>
+        /// This constant specifies that we want to use the default Run mode for the thread's Run loop.
+        /// </summary>
+        /// <remarks>
+        /// For more information, see the Apple documentation: https://developer.apple.com/library/mac/documentation/CoreFoundation/Reference/CFRunLoopRef/index.html
+        /// </remarks>
+        internal static IntPtr kCFRunLoopDefaultMode = CFStringCreateWithCString(IntPtr.Zero, "kCFRunLoopDefaultMode", 0);
+
+        /// <summary>
+        /// This is the default system path for the CoreFoundation library that PInvoke will use
+        /// </summary>
+        private const string CoreFoundationLibrary = "/System/Library/Frameworks/CoreFoundation.framework/CoreFoundation";
+
+        /// <summary>
+        /// Tells the OS what encoding the passed in String is in. These come from the CFString.h header file in the CoreFoundation framework.
+        /// </summary>
+        internal enum CFStringBuiltInEncodings : uint
```

Comment:
Gotcha, the CoreServices can probably be renamed to Interop.EventStream. The CoreFoundation stuff is all pretty related where if you call CFArrayCreate or CFStringCreate* then you'll need to call CFRelease on the pointer.

That being said, I can probably pull out the RunLoop interop into it's own file since it is separate from the String, Array, and Release code.